### ISSUE TITLE
Unable to connect to serviceMqttException, sometimes ...

### ISSUE THREAD

**Issue by ApiO** [Azure/azure-iot-sdk-java#126]
Hi,

## Environement

- Java:  jdk 1.7.0.40
- IDE: Neon.3 Release (4.6.3)
- Package version:
```
      <groupId>com.microsoft.azure.sdk.iot</groupId>
      <artifactId>iot-device-client</artifactId>
      <version>1.3.31</version>
```

## Description

Here are some feedback/questions:

Often, when I open a DeviceClient using mqtt protocol I have the following IO Exception:
```
java.io.IOException: Unable to connect to service
	at com.microsoft.azure.sdk.iot.device.transport.mqtt.MqttIotHubConnection.open(MqttIotHubConnection.java:140)
	at com.microsoft.azure.sdk.iot.device.transport.mqtt.MqttTransport.open(MqttTransport.java:83)
	at com.microsoft.azure.sdk.iot.device.DeviceIO.open(DeviceIO.java:212)
	at com.microsoft.azure.sdk.iot.device.DeviceClient.open(DeviceClient.java:197)
```

I've seen [this issue](https://github.com/Azure/azure-iot-sdk-java/issues/86) and I understand the TLS handshake mechanism but using the sdk I doesn't know what fails and it's a bit frustrating. Does server validation fail ? Client ? Is there a way to investigate or have more details ?

I've some "retry", but it's an horrible workaround : it' is terribly slow. Users can wait for years according to the successives timeout.

Now, I seek for more "stability" or simply reactivity. Is there something I can do ?

Note: no, I cannot change my depreciate version of the jdk (unfortunately).

**Comment by ApiO** [comment]
After looking into the sdk code I found out that my exceptions were thrown by:

Mqtt.class, com.microsoft.azure.sdk.iot.device.transport.mqtt, l.221
> connectToken.waitForCompletion();

With the internal message:
> Unable to connect to server (32103) - java.net.ConnectException: Connection timed out: connect

That is the kind of informations I was looking for. Why not send this level of information to the sdk user ?

**Comment by prmathur-microsoft** [comment]
You are right about bubbling up the exception to the user, a functionality that lacks in the SDK as we currently stand. We have this in our backlog and we are working to resolve this. In the meantime I will be happy to review your PR if you want to provide one.

**Comment by timtay-microsoft** [comment]
While I understand that you cannot upgrade from your version, this problem has been resolved as of [Device SDK release 1.6.0](https://github.com/Azure/azure-iot-sdk-java/releases/tag/2017-12-2)

I'm going to close this issue since exceptions are now properly bubbled up to the user. Feel free to re-open if something else comes up.
